### DISCUSSION TITLE
Possible Bug with subcomponents

### DISCUSSION THREAD

**Discussion by EthanKC** [nextgenhealthcare/connect#6180] [Q&A] [answered]
I wanted to discuss this here before submitting it as a formal issue (which admittedly isn't something I've ever done, so I'm a bit unsure about it). 

I've discovered that in version 4.5.0 (possibly others, just didn't discover this till now) that if you are populating subfields in an empty field, if the .2 transformer is before the .1 transformer, it actually populates the .1 value into the .3 subcomponent. For example, I had two message builder transformers, one to populate OBR.24.2 with "WCM", and the other to populate OBR.24.1 with "MG":
<img width="176" alt="image" src="https://github.com/nextgenhealthcare/connect/assets/100318963/fa80afd7-99b1-40cb-be01-e40e1bf8b7a4">

When I send a message with the transformers in the above order, with the OBR.24.2 before the OBR.24.1, it comes out with this result:
OBR|1|PRO198596||77066OT^3D OFFICE DIAGNOSTIC BILATERAL MAMMOGRAM ^L|||20240423155856||||N|||||1629323779^Miller^Jennifer||||||||**^WCM^MG**|||^^^^^R|1396748604^Bryan^David

But if I move the OBR.24.1 to before the OBR.24.2, it works correctly:

OBR|1|PRO198596||77066OT^3D OFFICE DIAGNOSTIC BILATERAL MAMMOGRAM ^L|||20240423155856||||N|||||1629323779^Miller^Jennifer||||||||**MG^WCM**|||^^^^^R|1396748604^Bryan^David

I tested this with a handful of OBR fields, and with a couple of PV1 fields, and it behaved the same for all of them. My understanding is that Mirth should recognize where the value should go regardless of the order of transformers. OBR.24.1 doesn't change just because the transformer to populate it comes after one that populates OBR.24.2. Am I missing something here?

**Comment by pacmano1** [comment, marked as answer]
Short answer - build stuff in order.  There are a handful of posts that discuss this same "problem" but not using the UI to build stuff.

'My understanding is that Mirth should recognize where the value should go regardless of the order of transformers."  No. it does not unless it exists already.

**Reply by pacmano1** [reply]
https://github.com/nextgenhealthcare/connect/discussions/4917, I think this is a related discussion.

**Reply by EthanKC** [reply]
Ok, good to know. Just wanted to make sure it was me misunderstanding the expectation, and not a bug. Thanks!

**Reply by tonygermano** [reply]
Building stuff in order is the best solution. If there is a reason why that can't be done, there is a code template found here that you can run at the end of your transformer to sort everything into the correct order. https://github.com/nextgenhealthcare/connect-examples/tree/master/Code%20Templates/Fix%20HL7%20Node%20Order

There are other useful code templates in that repository as well, so it's worth a browse.
